src/coreclr/vm/dllimportcallback.h

```diff
@@ -223,18 +263,32 @@ class UMEntryThunk
         }
         else
         {
-            if (m_pManagedTarget != (TADDR)0)
+            PTR_UMEntryThunkData data = GetData();
+            if (data->m_pManagedTarget != (TADDR)0)
             {
-                RETURN m_pManagedTarget;
+                RETURN data->m_pManagedTarget;
             }
             else
             {
-                RETURN m_pMD->GetMultiCallableAddrOfCode();
+                RETURN data->m_pMD->GetMultiCallableAddrOfCode();
             }
         }
     }
 #endif // !DACCESS_COMPILE
 
+    OBJECTHANDLE GetObjectHandleSpeculative() const
```

Comment:
What is the situation that the Speculative version allows that the non-speculative version does not allow?

Reply:
(It is not obvious to me why we need the speculative variant.)